tests/file_storage/tests.py

```diff
@@ -297,6 +298,18 @@ def test_file_save_with_path(self):
 
         self.storage.delete('path/to/test.file')
 
+    @unittest.skipUnless(symlinks_supported(), 'Must be able to symlink to run this test.')
+    def test_file_save_broken_symlink(self):
+        """
+        A new path is created on save when a broken symlink is supplied.
+        """
+        nonexistent_file_path = os.path.join(self.temp_dir, 'nonexistent.txt')
+        broken_symlink_path = os.path.join(self.temp_dir, 'symlink.txt')
+        os.symlink(nonexistent_file_path, broken_symlink_path)
+        f = ContentFile('some content')
+        f_name = self.storage.save(broken_symlink_path, f)
+        self.assertNotIn('nonexistent.txt', self.storage.path(f_name))
```

Comment:
We want to save file with the `symlink.txt` name so this assertion doesn't have much value IMO. I would check that the new file exists.